packages/account/imAccount/validators/ecdsaValidator.ts

```diff
@@ -0,0 +1,86 @@
+import * as ethers from "ethers"
+
+import ECDSAValidatorMetadata from "~packages/account/imAccount/abis/ECDSAValidator.json"
+import type { Validator } from "~packages/account/imAccount/validator"
+import type { HexString } from "~typing"
+
+export class ECDSAValidator implements Validator {
+  private node: ethers.JsonRpcProvider
+  private owner: ethers.Wallet
+  public contract: ethers.Contract
+
+  public constructor(opts: {
+    address: HexString
+    ownerPrivateKey: string
+    nodeRpcUrl: string
+  }) {
+    this.node = new ethers.JsonRpcProvider(opts.nodeRpcUrl)
+    this.owner = new ethers.Wallet(opts.ownerPrivateKey)
+    this.contract = new ethers.Contract(
+      opts.address,
+      ECDSAValidatorMetadata.abi,
+      this.node
+    )
+  }
+
+  public async sign(message: string | Uint8Array): Promise<HexString> {
```

Comment:
nit: Use `BytesLike` from `~typing`
```suggestion
  public async sign(message: BytesLike): Promise<HexString> {
```